selfdescribe.json

```diff
@@ -3666,7 +3666,7 @@
+      "doc": "This monitor lets you provide custom collectd\nconfiguration to be run by the managed collectd instance.  You can provide\nconfiguration for as many plugins as you want in a single instance of this\nmonitor configuration by either putting multiple `\u003cPlugin\u003e` blocks in a\nsingle `template` option, or specifying multiple `templates`.\n\nNote that a distinct instance of collectd is run for each instance of this\nmonitor, so it is more efficient to group plugin configurations into a\nsingle monitor configuration (either in one big `template` text blob, or\nsplit into multiple `templates`).  You should not group configurations if\nusing a discoveryRule since that would result in duplicate config for each\ninstance of the service endpoint discovered.\n\nYou can also use your own Python plugins in conjunction with the\n`ModulePath` option in\n[collectd-python](https://collectd.org/documentation/manpages/collectd-python.5.shtml).\nIf your Python plugin has dependencies of its own, you can specify the path\nto them by specifying multiple `ModulePath` options with those paths.\n\nHere is an example of a configuration with a custom Python plugin:\n\n```yaml\n  - type: collectd/custom\n    discoveryRule: container_image =~ \"myservice\"\n    template: |\n      LoadPlugin \"python\"\n      \u003cPlugin python\u003e\n        ModulePath \"/usr/lib/python2.7/dist-packages/health_checker\"\n        Import \"health_checker\"\n        \u003cModule health_checker\u003e\n          URL \"http://{{.Host}}:{{.Port}}\"\n          JSONKey \"isRunning\"\n          JSONVal \"1\"\n        \u003c/Module\u003e\n      \u003c/Plugin\u003e\n```\n\nWe have many collectd plugins included in the image that are not exposed as\nmonitors.  You can see the plugins in the `\u003cAGENT_BUNDLE\u003e/lib/collectd`\ndirectory, where `\u003cAGENT_BUNDLE\u003e` is the root of the filesystem in the\ncontainerized version, and is normally `/usr/lib/signalfx-agent` in the\nnon-containerized agent.\n\n### Running the collectd/exec plugin\nIt is possible to use the collectd/custom monitor to run the collectd/exec\nplugin. If you are running the agent in a container, use the `signalfx-agent` user\nwhen to execute the script.\n```yaml\n  - type: collectd/custom\n    template: |\n      LoadPlugin exec\n      \u003cPlugin exec\u003e\n        Exec \"signalfx-agent\" \"/path/to/script.sh\"\n      \u003c/Plugin\u003e\n```\nThe [collectd-exec](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_exec)\ndocumentation outlines how to use the plugin.\n",
```

Comment:
You can use this monitor to collect and run custom collectd\nconfiguration data on managed collectd instances. You can collect\nconfiguration data for multiple plugins for a single instance of this\nmonitor configuration. To do so, put multiple `\u003cPlugin\u003e` blocks in a \nsingle ‘template’ option. Or, you can add the block in multiple `templates`.\n\Remember that a unique instance of collectd runs for each instance of this\nmonitor. As a result, it is more efficient to group plugin configurations into a \nsingle monitor configuration. To accomplish this, you can use one large `template` text blob or\nsplit into multiple `templates`. Do not group configurations if \nusing is a discoveryRule because duplicate configs for each \ninstance of the service endpoint will be discovered.\n\nYou can also use your own Python plugins along with the\n`ModulePath` option in\n[collectd-python](https://collectd.org/documentation/manpages/collectd-python.5.shtml).\nIf your Python plugin has dependencies, then you can specify the path\nto the dependencies by specifying multiple `ModulePath` options with those paths.\n\nReview the following example of a configuration with a custom Python plugin:

Reply:
@galpizarsfx this is just an auto-generated blob from the metadata.yaml file above.  You can just make your suggestions on that and there is a CI job to make sure the developer has properly updated the rest of the docs when pushing.